Makefile

```diff
@@ -417,6 +417,9 @@ proto-update-deps:
 	@mkdir -p $(COSMOS_TYPES)/crypto/multisig/v1beta1
 	@curl -sSL $(COSMOS_URL)/crypto/multisig/v1beta1/multisig.proto > $(COSMOS_TYPES)/crypto/multisig/v1beta1/multisig.proto
 
+	@mkdir -p $(COSMOS_TYPES)/authz/v1beta1
+	@curl -sSL $(COSMOS_URL)/authz/v1beta1/authz.proto > $(COSMOS_TYPES)/authz/v1beta1/authz.proto
```

Comment:
It is used in `MsgGrantTreasuryAuthorization` [here](https://github.com/desmos-labs/desmos/blob/095e5278ec7336f029838612efb612c764d87497/proto/desmos/subspaces/v3/msgs_treasury.proto#L23). I applied it since `Grant` object provides a set of [methods](https://github.com/cosmos/cosmos-sdk/blob/v0.45.11/x/authz/authorization_grant.go#L12) like `ValidateBasic`, which are useful to reduce codes and already tested in cosmos-sdk.

Reply:
For example, we can simply use `Grant.GetAuthorization` when saving it by [`SaveGrant`](
https://github.com/desmos-labs/desmos/blob/095e5278ec7336f029838612efb612c764d87497/x/subspaces/keeper/msg_server_treasury.go#L34) to replace the code as follows:
```go
authorization, ok := msg.Authorization.GetCachedValue().(authz.Authorization)
if !ok {
    return nil, fmt.Errorof("invalid authorization")
}
err = k.authzk.SaveGrant(ctx, grantee, treasury, authorization, msg.Expiration)
if err != nil {
    return nil, err
}
```